tools/tsp-client/src/index.ts

```diff
@@ -84,7 +89,12 @@ async function sdkInit(
       throw new Error(`tspconfig.yaml is empty at ${config}`);
     }
     const configYaml = parseYaml(data);
-    const serviceDir = configYaml?.parameters?.["service-dir"]?.default;
+    // Check if service-dir is defined in the emitter specific configurations in tspconfig.yaml
```

Comment:
maybe you could encapsulate this into its own function that you can share in the 2 places where this is being done

Reply:
This is really personal preference but wanted to share in case you find it useful. You could do this conditional assignment in a single line
```ts
// Check if service-dir is defined in the emitter specific configurations in tspconfig.yaml
// Default to the top level service-dir parameter in tspconfig.yaml
const serviceDir = configYaml?.options?.[emitter]?.["service-dir"] ?? configYaml?.parameters?.["service-dir"]?.default;
```